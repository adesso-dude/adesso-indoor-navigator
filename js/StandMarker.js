import React from 'react';
import PropTypes from 'prop-types';

import {
  StyleSheet,
  View,
  Text,
} from 'react-native';

const propTypes = {
  name: PropTypes.string.isRequired,
  fontSize: PropTypes.number,
};

const defaultProps = {
  fontSize: 8,
};

class StandMarker extends React.Component {
  render() {
    const { fontSize, name } = this.props;
    return (
      <View style={styles.container}>
        <View style={styles.bubble}>
          <Text style={[styles.amount, { fontSize }]}>{name}</Text>
        </View>
        <View style={styles.arrowBorder} />
        <View style={styles.arrow} />
      </View>
    );
  }
}

StandMarker.propTypes = propTypes;
StandMarker.defaultProps = defaultProps;

const styles = StyleSheet.create({
  container: {
    flexDirection: 'column',
    alignSelf: 'flex-start',
  },
  bubble: {
    flex: 0,
    flexDirection: 'row',
    alignSelf: 'flex-start',
    backgroundColor: '#FF5A5F',
    padding: 2,
    borderRadius: 3,
    borderColor: '#D23F44',
    borderWidth: 0.5,
  },
  dollar: {
    color: '#FFFFFF',
    fontSize: 10,
  },
  amount: {
    color: '#FFFFFF',
    fontSize: 13,
  },
  arrow: {
    backgroundColor: 'transparent',
    borderWidth: 4,
    borderColor: 'transparent',
    borderTopColor: '#FF5A5F',
    alignSelf: 'center',
    marginTop: -9,
  },
  arrowBorder: {
    backgroundColor: 'transparent',
    borderWidth: 4,
    borderColor: 'transparent',
    borderTopColor: '#D23F44',
    alignSelf: 'center',
    marginTop: -0.5,
  },
});

export default StandMarker;
