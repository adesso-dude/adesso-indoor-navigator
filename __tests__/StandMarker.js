import 'react-native';
import React from 'react';
import StandMarker from '../js/StandMarker';

// Note: test renderer must be required after react-native.
import renderer from 'react-test-renderer';

jest.mock('react-native');

describe('<StandMarker>', () => {
    it('renders correctly', () => {
        const tree = renderer.create(
          <StandMarker name={'Test'} />
        ).toJSON();

        expect(tree).toMatchSnapshot();
    });

    it('Should have a fontSize of 8 if none is set', () => {
        const tree = renderer.create(
            <StandMarker name={'Test'} />
        );
    });
});