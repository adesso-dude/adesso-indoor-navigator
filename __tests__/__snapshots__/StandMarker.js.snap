// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<StandMarker> renders correctly 1`] = `
<View
  style={
    Object {
      "alignSelf": "flex-start",
      "flexDirection": "column",
    }
  }
>
  <View
    style={
      Object {
        "alignSelf": "flex-start",
        "backgroundColor": "#FF5A5F",
        "borderColor": "#D23F44",
        "borderRadius": 3,
        "borderWidth": 0.5,
        "flex": 0,
        "flexDirection": "row",
        "padding": 2,
      }
    }
  >
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Array [
          Object {
            "color": "#FFFFFF",
            "fontSize": 13,
          },
          Object {
            "fontSize": 8,
          },
        ]
      }
    >
      Test
    </Text>
  </View>
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderTopColor": "#D23F44",
        "borderWidth": 4,
        "marginTop": -0.5,
      }
    }
  />
  <View
    style={
      Object {
        "alignSelf": "center",
        "backgroundColor": "transparent",
        "borderColor": "transparent",
        "borderTopColor": "#FF5A5F",
        "borderWidth": 4,
        "marginTop": -9,
      }
    }
  />
</View>
`;
